src/keys-builder/typescript/build-keys-from-ast-nodes.ts

```diff
@@ -27,25 +28,33 @@ export function buildKeysFromASTNodes(
       if (allowedMethods.includes(methodName) === false) {
         continue;
       }
-      const data = {};
 
-      const [key, _, lang] = node.parent.arguments;
-      if (isStringLiteral(key) || isNoSubstitutionTemplateLiteral(key)) {
-        data['key'] = key.text;
-      }
+      const [keyNode, _, langNode] = node.parent.arguments;
+      let lang: string;
+      let keys: string[] = [];
 
-      if (!data['key']) continue;
+      if (isStringNode(langNode)) {
+        lang = langNode.text;
+      }
 
-      if (
-        lang &&
-        (isStringLiteral(lang) || isNoSubstitutionTemplateLiteral(lang))
-      ) {
-        data['lang'] = lang.text;
+      if (isStringNode(keyNode)) {
+        keys = [keyNode.text];
+      } else if (isArrayLiteralExpression(keyNode)) {
+        keys = keyNode.elements.filter(isStringNode).map((node) => node.text);
       }
 
-      result.push(data);
+      for (const key of keys) {
+        const data = lang ? { lang, key } : { key };
+        result.push(data);
+      }
     }
   }
 
   return result;
 }
+
+function isStringNode(node: Node) {
```

Comment:
```suggestion
function isStringNode(node?: Node) {
```

Reply:
also, do i need to run `npm run commit` for each update?

Reply:
as I said before, you can commit several changes together, but in the end I will want you to squash all the commits into one and force push to your branch.

Regarding the optional you can leave it as is I thought that lang can be undefined but I just checked the types.